### PR TITLE
Feature: Save act of service pledge and heart success screen (kids-1507)

### PR DESCRIPTION
**Pull request by Daniela510**
## Summary by CodeRabbit

- **New Features**
	- Introduced a new Lottie animation for success feedback.
	- Added a new parameter for conditional rendering in organization-related widgets.
	- Implemented functionality to save pledges for organizations and navigate to a success screen after completing acts of service.

- **Bug Fixes**
	- Updated method names for clarity and consistency.

- **Documentation**
	- Enhanced descriptions for new methods and parameters related to the grateful functionality.

### CODE REVIEW

File: lib/features/family/features/reflect/presentation/pages/heart_success_screen.dart [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+            Container(
+              padding: const EdgeInsets.all(40),
+              width: double.infinity,
+              child: Column(
+                children: [
+                  TitleLargeText(
+                    organisation.thankYou ?? 'Awesome!',
+                    textAlign: TextAlign.center,
+                  ),
+                  Text(
```
Comment (TammiLion) [inline lib/features/family/features/reflect/presentation/pages/heart_success_screen.dart:73]: Just to be sure, did the design change? I would not expect the word donation here? In the "old" figma i see "Good luck in completing your act of service".

**Review by TammiLion — APPROVED** [review]
1 question/comment

File: lib/features/family/features/reflect/bloc/gratitude_selection_cubit.dart [hunk 2]
```diff
@@ -40,6 +40,8 @@ class GratitudeSelectionCubit
       GratitudeSelectionUimodel(
         gratitudeList: gratitudeCategories,
         selectedGratitude: getSelectedCategory(),
+        superheroName:
+            _reflectAndShareRepository.getCurrentSuperhero().firstName,
```
[the remark this answers is not in the document]
Comment (Daniela510) [reply lib/features/family/features/reflect/bloc/gratitude_selection_cubit.dart:44]: this is for tracking purposes and not relevant

File: lib/features/family/features/reflect/presentation/pages/heart_success_screen.dart (hunk 1 above)
Comment (Daniela510) [reply lib/features/family/features/reflect/presentation/pages/heart_success_screen.dart:73]: right, i was looking at the lottie flow it doesnt have words, and thought the words were the same

**Pull request merged**